IronPython3Extension/Extension.cs

```diff
@@ -29,9 +35,78 @@ public void Dispose()
             
         }
 
-        public void Ready(ReadyParams sp)
+        /// <summary>
+        /// Action to be invoked when the Dynamo has started up and is ready
+        /// for user interaction. 
+        /// </summary>
+        /// <param name="rp"></param>
+        public void Ready(ReadyParams rp)
+        {
+            var extraPath = Path.Combine(new FileInfo(Assembly.GetExecutingAssembly().Location).Directory.Parent.FullName, "extra");
+            var alc = new IsolatedPythonContext(Path.Combine(extraPath, $"{PythonEvaluatorAssembly}.dll"));
+            var dsIronAssem = alc.LoadFromAssemblyName(new AssemblyName(PythonEvaluatorAssembly));
+
+            //load the engine into Dynamo ourselves.
+            LoadPythonEngine(dsIronAssem);
+
+            //we used to do this:
+            //but it's not neccesary to load anything into the VM.
+            //instead we skip all the extra work and trigger the side effect we want
+            //which is re executing the graph after the dsIronPython evaluator is loaded into the PythonEngineManager.
+            //rp.StartupParams.LibraryLoader.LoadNodeLibrary(dsIronAssem);
+
+            if (rp.CurrentWorkspaceModel is HomeWorkspaceModel hwm)
+            {
+                foreach (var n in hwm.Nodes)
+                {
+                    n.MarkNodeAsModified(true);
+                }
+                hwm.Run();
+            }
+        }
+
+        private static void LoadPythonEngine(Assembly assembly)
         {
+            if (assembly == null)
+            {
+                return;
```

Comment:
this would be very weird. It would mean that IRonpython will not be loaded.
Should we rather throw an exception ? to be caught by Dynamo ...

Reply:
yes, throwing seems better in the context of an extension - I can just get rid of this if  statement and let the try catch below throw.